### PR TITLE
Automatic active space selection with SCINE autoCAS

### PR DESCRIPTION
**Pull request by AlexandreF-1qbit**
Script to perform an active space search calculation with [SCINE autoCAS](https://github.com/qcscine/autocas/tree/master). The example shows an example with N2 / cc-pVDZ.

### CODE REVIEW

File: examples/chemistry/autocas.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+import h5py
+import numpy as np
+import yaml
+
+from scine_autocas.main_functions import MainFunctions
+from tangelo import SecondQuantizedMolecule
+
+
+def call_autocas(sqmol: SecondQuantizedMolecule, settings: dict) -> dict:
+    """Calls SCINE autoCAS to perform active space calculation on a molecule.
```
Comment (JamesB-1qbit) [inline examples/chemistry/autocas.py:50]: Calls SCINE autoCAS to generate the active space of a molecule

File: examples/chemistry/autocas.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        "yaml_input": yaml_file
+    })
+
+    return autocas_workflow.results
+
+
+def apply_active_space(sqmol: SecondQuantizedMolecule, autocas_results: dict) -> None:
+    """Applies the active space selection obtained from autoCAS results to the
+    molecule. Modifies the 'sqmol' object in place by freezing molecular
+    orbitals based on autoCAS results. Does not return any value.
```
Comment (JamesB-1qbit) [inline examples/chemistry/autocas.py:96]: Applies the active space obtained from autoCAS to the molecule.
Modifies the 'sqmol' object by in-place freezing molecular orbitals and utilizing the molecular
coefficients returned from autoCAS. Does not return any value.

**Review by JamesB-1qbit — CHANGES_REQUESTED** [review]
Small docstring suggestions.

File: examples/chemistry/autocas.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        N 0.0000 0.0000 1.0900
+    """
+    n2 = SecondQuantizedMolecule(xyz_n2, q=0, spin=0, basis="cc-pvdz")
+
+    # Call the autocas workflow, and apply the active space selection.
+    autocas_results = call_autocas(n2, settings)
+    apply_active_space(n2, autocas_results)
+
+    # Print the new molecule object.
+    print(n2)
```
Comment (ValentinS4t1qbit) [inline examples/chemistry/autocas.py:146]: Probably a good way to close is to state the molecule object is ready to be pushed in any classical or quantum solver.
Maybe we could run FCI on this to show the energy and state the AS has indeed been applied.

**Review by ValentinS4t1qbit — APPROVED** [review]

**Pull request merged**